### ISSUE TITLE
Empty interpolation of schema.Set values

### ISSUE THREAD

**Issue by vancluever** [hashicorp/terraform#10977]
I didn't see anything in the TF changelog that looked inparticularly damning on this one, but this definitely seems like a regression between v0.8.1. and v0.8.2:

## Code

```
data "aws_alb" "autoscaling_alb" {
  arn = "arn.goes.here"
}

output "foo" {
  value = "${data.aws_alb.autoscaling_alb.security_groups}"
}

```

## v0.8.1

```
Outputs:

foo = [
    sg-abcdef12
]
```

## v0.8.2

```
Outputs:

foo = [
    
]
```

## Other Curiosities

The lookup result is not a bad index, but a literally blank string. If you do this on sets that contain multiple items (the ALB I was looking at only had 1 security group, you actually get something like:

```
foo = [
    ,
    
]
```

What does work is looking up the set item directly via its hash. So if you were to do something like `value = "${data.aws_alb.autoscaling_alb.security_groups.123456789}"`, that would work.

Going to keep digging and see what I can find.

**Comment by vancluever** [comment]
Think I found it:

https://github.com/hashicorp/terraform/blob/master/flatmap/expand.go#L39

`expandArray` (which complex value interpolation was switched to a few weeks ago on part of that functionality moving to `flatmap.Expand`) does not take into account list structures that are actually of type `schema.Set`, because it bases its index lookups off a sequence of `0` - `len-1` of the attribute. Since the "keys" for the set are actually hashed, you can be pretty much always sure that you will not get a proper lookup here.

**Comment by vancluever** [comment]
I think I got a fix in for this! Commit references back to this issue

**Comment by mitchellh** [comment]
Will be fixed by #11042
